Wasmswap/wasmswap-interface
# Intelligent swap routes

This is to introduce intelligent swap routes in the swap module and token price queries. It will now query price for every possible swap route to figure out the cheapest way for a swap.

### Steps to test
- In `useTokenSwap` hook, log the `tokenToTokenPrice` variable. The variable includes swap price plus selected swap route.
- Choose a pair for swap that has two possible routes; pass through swap & direct pool token swap;
- First, make sure there's enough liquidity in the direct pool for that given token pair; Then swap the tokens to validate if that exact pool was chosen for the swap;
- Then remove liquidity from the direct token pair pool and try to do the same swap;
- Observe if the pass through swap route was selected for the swap.
